op-challenger/game/keccak/merkle/tree_test.go

```diff
@@ -0,0 +1,214 @@
+package merkle
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/keccak/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBinaryMerkleTree_AddLeaf(t *testing.T) {
+	var tests []struct {
+		name      string
+		leafCount int
+	}
+
+	// Test only three leaf counts since this requires adding a lot of leaves.
+	// To test more thoroughly, increase the divisor and run locally.
+	for i := 0; i < MaxLeafCount; i += MaxLeafCount / 3 {
+		tests = append(tests, struct {
+			name      string
+			leafCount int
+		}{
+			name:      fmt.Sprintf("AddLeaf-%d", i),
+			leafCount: i,
+		})
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tree := NewBinaryMerkleTree()
+			expectedLeafHash := zeroHashes[BinaryMerkleTreeDepth-1]
+			for i := 0; i < test.leafCount; i++ {
+				input := ([136]byte)(bytes.Repeat([]byte{byte(i)}, 136))
```

Comment:
`types.BlockSize` instead of hardcoding 136? Same comment for all mentions of 136.

Reply:
Oops I missed those since I had some changes stashed from before your type cleanup pr, fixing rn. Good catch!